packages/data/src/redux-store/index.js

```diff
@@ -197,25 +197,31 @@ export default function createReduxStore( key, options ) {
 				} )
 			);
 
-			let selectors = mapSelectors(
-				{
-					...mapValues(
-						metadataSelectors,
-						( selector ) =>
-							( state, ...args ) =>
-								selector( state.metadata, ...args )
-					),
-					...mapValues( options.selectors, ( selector ) => {
-						if ( selector.isRegistrySelector ) {
-							selector.registry = registry;
-						}
+			function bindSelector( selector ) {
+				if ( selector.isRegistrySelector ) {
+					selector.registry = registry;
+				}
+				const boundSelector = ( ...args ) => {
+					const state = store.__unstableOriginalGetState();
+					return selector( state.root, ...args );
+				};
+				boundSelector.hasResolver = false;
+				return boundSelector;
```

Comment:
Given that this part is the same as `bindMetadataSelector()`, the only difference being where the selector reads state from, I wonder if we could further abstract this fragment into a separate function that receives the state tree name. Then we could reuse that in both `bindSelector` and `bindMetadataSelector`.

Reply:
I originally did this, if you look at the implementation in #51051, there are calls `bindSelector( 'root' )` and `bindSelector( 'metadata' )`.

But now I plan to diverge the two versions even more. The metadata selectors only need the binding, we cal remove the `hasResolver` part. And when binding the classic selector, I would like to try merging the resolver fulfillment also into `bindSelector`. So that there is only one wrapper and when in debugger, you step into a call like:
```js
registry.select( 'core/block-editor' ).getBlocks();
```
you are just one step away from the actual implementation of `getBlocks`. Until recently, you'd have to step through several layers of wrappers.